src/index.ts

```diff
@@ -130,10 +143,48 @@ export function getInstance(): EppoClient {
   return clientInstance;
 }
 
-function newEventDispatcher(sdkKey: string) {
-  const eventQueue = new FileBackedNamedEventQueue<Event>('events');
+function newEventDispatcher(
+  sdkKey: string,
+  config: IClientConfig['eventIngestionConfig'] = {},
+): EventDispatcher {
+  const {
+    batchSize = 1_000,
+    deliveryIntervalMs = 10_000,
+    retryIntervalMs = 5_000,
+    maxRetryDelayMs = 30_000,
+    maxRetries = 3,
+    maxQueueSize = 10_000,
+    disabled = false,
+  } = config;
+  if (disabled) {
+    return NO_OP_EVENT_DISPATCHER;
+  }
+  let eventQueue: NamedEventQueue<Event>;
+  try {
+    // Check if the file system is read-only
+    if (isReadOnlyFs(`${process.cwd()}/.queues`)) {
+      applicationLogger.warn(
+        'File system appears to be read-only. Using in-memory event queue instead.',
+      );
+      eventQueue = new BoundedEventQueue<Event>('events', [], maxQueueSize);
+    } else {
+      eventQueue = new FileBackedNamedEventQueue<Event>('events');
+    }
+  } catch (error) {
+    // If there's any error during the check, fall back to BoundedEventQueue
+    applicationLogger.warn(
+      `Error checking file system: ${error}. Using in-memory event queue instead.`,
+    );
+    eventQueue = new BoundedEventQueue<Event>('events', [], maxQueueSize);
+  }
```

Comment:
Perhaps a `finally` if `BoundedEventQueue` is somehow the source of the thrown error? Although maybe that's impossible